src/data_types.js

```diff
@@ -220,7 +220,7 @@ class DATE extends DataType {
   constructor(precision, timezone = true) {
     super();
     this.dataType = 'datetime';
-    this.precision = precision;
+    this.precision = precision || 0;
```

Comment:
precision should not default to 0. At least the timestamp precision is not default to 0 in neither mysql nor postgresql.